### PR TITLE
Features/roc 8086 pcq integration

### PR DESCRIPTION
**Pull request by pradhap87**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-8086
https://tools.hmcts.net/jira/browse/ROC-8085

### Change description

PCQI || Invoking PCQ Service from CMC & Saving the PCQID in CCD

### Work checklist

- [x] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [x] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by prudhvi-v** [comment]
Can you just check that 'code Smell' comment in the sonarclound. Other than that I am ok with the changes.

**Comment by pradhap87** [comment]
> Can you just check that 'code Smell' comment in the sonarclound. Other than that I am ok with the changes.

I have updated it

**Comment by sunilkumar-cts** [comment]
Can you check attached unit tests? There are couple of simple lines to cover, we need to cover all simple things in order to have higher average, else they build up later when we have complex code.

File: config/default.json [hunk 1]
```diff
@@ -93,7 +93,8 @@
     "webChat": true,
     "inversionOfControl": true,
     "warningBanner": false,
-    "helpWithFees": false
+    "helpWithFees": false,
+    "pcq":true
```
[the remark this answers is not in the document]
Comment (pradhap87) [reply config/default.json:97]: This was a merge from the master branch (not a PCQ change), HWF has the feature toggle in place. We no need to worry about this value.

File: src/test/features/response/routes/check-and-send.ts [hunk 2]
```diff
@@ -93,7 +93,14 @@ describe('Defendant response: check and send page', () => {
           await request(app)
             .get(pagePath)
             .set('Cookie', `${cookieName}=ABC`)
-            .expect(res => expect(res).to.be.successful.withText('Check your answers'))
+            .expect(res => {
+              if (FeatureToggles.isEnabled('pcq')) {
+                res.status = 302
```
[the remark this answers is not in the document]
Comment (pradhap87) [reply src/test/features/response/routes/check-and-send.ts]: I have updated (expect(res.status).to.be.equal(302))

File: src/test/features/claim/routes/evidence.ts [hunk 3]
```diff
@@ -63,7 +64,13 @@ describe('Claim issue: evidence', () => {
           .post(pagePath)
           .set('Cookie', `${cookieName}=ABC`)
           .send({ rows: [{ type: EvidenceType.CONTRACTS_AND_AGREEMENTS.value, description: 'Bla bla' }] })
-          .expect(res => expect(res).to.be.redirect.toLocation(Paths.taskListPage.uri))
+          .expect(res => {
+            if (FeatureToggles.isEnabled('pcq')) {
+              res.status = 302
```
[the remark this answers is not in the document]
Comment (pradhap87) [reply src/test/features/claim/routes/evidence.ts]: I have updated (expect(res.status).to.be.equal(302))

File: src/test/features/claim/routes/evidence.ts [hunk 4]
```diff
@@ -79,7 +86,13 @@ describe('Claim issue: evidence', () => {
               description: undefined
             }]
           })
-          .expect(res => expect(res).to.be.redirect.toLocation(Paths.taskListPage.uri))
+          .expect(res => {
+            if (FeatureToggles.isEnabled('pcq')) {
+              res.status = 302
```
[the remark this answers is not in the document]
Comment (pradhap87) [reply src/test/features/claim/routes/evidence.ts]: I have updated (expect(res.status).to.be.equal(302))

**Comment by pradhap87** [comment]
Jenkins pipeline is getting failed due to dynamic AKS env build failure. So the AKS dynamic environment is not deployed.

**Comment by pradhap87** [comment]
Deleting theis PR Due to build issue.

**Pull request closed without merging**